src/main/java/bio/terra/service/filedata/azure/util/BlobContainerCopierBuilder.java

```diff
@@ -0,0 +1,141 @@
+package bio.terra.service.filedata.azure.util;
+
+import com.azure.storage.blob.BlobUrlParts;
+import java.time.Duration;
+import java.util.List;
+import java.util.Objects;
+import org.apache.commons.lang.StringUtils;
+import org.apache.logging.log4j.util.Strings;
+
+/** Builder for the {@link BlobContainerCopier} */
+public final class BlobContainerCopierBuilder {
+
+  public static final int DEFAULT_LIST_OPERATION_TIMEOUT_IN_SECONDS = 30;
+  public static final int DEFAULT_POLLING_INTERVAL_IN_SECONDS = 5;
+  private static final int MIN_POLLING_INTERVAL = 2;
+  private static final int MIN_LIST_TIMEOUT = 10;
+  private static final String EMPTY_PREFIX = "";
```

Comment:
I don't think there's any value in creating a constant for the empty string. I would write `""` directly in the code where an empty string is needed.

Reply:
Hmm, the intent here is to explicitly indicate that by default an empty prefix is used and I believe this makes the intent clearer:

```java
private String sourceContainerPrefix = EMPTY_PREFIX;
```